plugin.video.dailymotion_com/default.py

```diff
@@ -132,19 +147,23 @@ def sortVideos1(url):
         url = urlMain+"/group/"+gid+"/videos?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1"
     else:
         url = urlMain+"/videos?fields=description,duration,id,owner.username,taken_time,thumbnail_large_url,title,views_total&"+type+"="+gid+"&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1"
-    addDir(translation(30015), url.replace("sort=recent", "sort=trending"), 'listVideos', "")
-    addDir(translation(30008), url, 'listVideos', "")
-    addDir(translation(30009), url.replace("sort=recent", "sort=visited"), 'sortVideos2', "")
+    addDir(translation(30015), url.replace("sort=recent", "sort=trending"), 'listVideos', _ipath + "trending.png")
+    addDir(translation(30008), url, 'listVideos', _ipath + "most_recent.png")
+    addDir(translation(30009), url.replace("sort=recent", "sort=visited"), 'sortVideos2', _ipath + "most_viewed.png")
     if type == "owner":
-        addDir("- "+translation(30038), urlMain+"/user/"+gid+"/playlists?fields=id,name,videos_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listUserPlaylists', "")
+        addDir("- "+translation(30038), urlMain+"/user/"+gid+"/playlists?fields=id,name,videos_total&sort=recent&limit="+itemsPerPage+"&family_filter="+familyFilter+"&localization="+language+"&page=1", 'listUserPlaylists', _ipath + "playlists.png")
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    xbmc.executebuiltin('Container.SetViewMode(500)')
     xbmcplugin.endOfDirectory(pluginhandle)
 
 def sortVideos2(url):
-    addDir(translation(30010), url.replace("sort=visited", "sort=visited-hour"), "listVideos", "")
-    addDir(translation(30011), url.replace("sort=visited", "sort=visited-today"), "listVideos", "")
-    addDir(translation(30012), url.replace("sort=visited", "sort=visited-week"), "listVideos", "")
-    addDir(translation(30013), url.replace("sort=visited", "sort=visited-month"), "listVideos", "")
-    addDir(translation(30014), url, 'listVideos', "")
+    addDir(translation(30010), url.replace("sort=visited", "sort=visited-hour"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30011), url.replace("sort=visited", "sort=visited-today"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30012), url.replace("sort=visited", "sort=visited-week"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30013), url.replace("sort=visited", "sort=visited-month"), "listVideos", _ipath + "most_viewed.png")
+    addDir(translation(30014), url, 'listVideos', _ipath + "most_viewed.png")
+    xbmcplugin.setContent(pluginhandle, 'addons')
+    xbmc.executebuiltin('Container.SetViewMode(500)')
```

Comment:
This isn't allowed.

Reply:
Corrected it to use only if user setup the relevant modes in settings